### PR TITLE
TRAN-8309: Add route icon

### PR DESCRIPTION
**Pull request by prafullint**
### Description
Add route icon field in the `njt-rails` and `cta-subway` module.

### Describe your changes (features, changes, fixes, etc.)
Update intersection proto file
Update factories 
Added checks on `njt-rails` and `cta-subway`

### CODE REVIEW

File: gtfs_realtime_translators/bindings/intersection_pb2.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # -*- coding: utf-8 -*-
 # Generated by the protocol buffer compiler.  DO NOT EDIT!
 # source: intersection.proto
-"""Generated protocol buffer code."""
+
+import sys
+_b=sys.version_info[0]<3 and (lambda x:x) or (lambda x:x.encode('latin1'))
```
Comment (joshverma) [inline gtfs_realtime_translators/bindings/intersection_pb2.py:6]: Was this line autogenerated?

File: gtfs_realtime_translators/translators/cta_subway.py [hunk 2]
```diff
@@ -51,6 +51,8 @@ def __make_trip_update(cls, _id, prediction):
                                                 parsed_prediction_time)
         scheduled_interval = cls.__get_scheduled_interval(is_scheduled,
                                                           prediction['schInt'])
+        
+        route_icon = cls.__get_route_icon(prediction['flags'], prediction['destNm'])
```
Comment (joshverma) [inline gtfs_realtime_translators/translators/cta_subway.py:55]:
```suggestion
        route_icon = cls.__get_route_icon(prediction['flags'], headsign)
```

File: gtfs_realtime_translators/translators/cta_subway.py [hunk 3]
```diff
@@ -75,3 +78,11 @@ def __get_scheduled_interval(cls, is_scheduled, scheduled_interval):
             scheduled_interval_seconds = int(scheduled_interval) * 60
             return scheduled_interval_seconds
         return None
+    
+    @classmethod
+    def __get_route_icon(cls, flags, destination):
```
Comment (joshverma) [inline gtfs_realtime_translators/translators/cta_subway.py:83]:
```suggestion
    def __get_route_icon(cls, flags, headsign):
```

File: gtfs_realtime_translators/translators/cta_subway.py [hunk 4]
```diff
@@ -75,3 +78,11 @@ def __get_scheduled_interval(cls, is_scheduled, scheduled_interval):
...
+        if flags == 'H':
```
Comment (joshverma) [inline gtfs_realtime_translators/translators/cta_subway.py:84]: We need to ignore case when checking the value.

```
if flags and flags.lower() == 'h'
```

File: gtfs_realtime_translators/translators/cta_subway.py [hunk 5]
```diff
@@ -75,3 +78,11 @@ def __get_scheduled_interval(cls, is_scheduled, scheduled_interval):
...
+            return 'holiday' 
+        if destination in ["Midway", "O'Hare"]:
```
Comment (joshverma) [inline gtfs_realtime_translators/translators/cta_subway.py:86]: We need to ignore case here, too.

```
if headsign and headsign.lower() in ["midway", "o'hare"]:
```

File: gtfs_realtime_translators/translators/njt_rail.py [hunk 6]
```diff
@@ -61,6 +61,7 @@ def __make_trip_updates(cls, data):
                         departure_time = int(scheduled_datetime.add(seconds=int(item_entry['SEC_LATE'])).timestamp())
                         scheduled_departure_time = int(scheduled_datetime.timestamp())
                         custom_status = item_entry['STATUS']
+                        route_icon = cls.__get_route_icon(item_entry['DESTINATION'])
```
Comment (joshverma) [inline gtfs_realtime_translators/translators/njt_rail.py:64]:
```suggestion
                        route_icon = cls.__get_route_icon(headsign)
```

File: gtfs_realtime_translators/translators/njt_rail.py [hunk 7]
```diff
@@ -181,3 +183,11 @@ def __get_route_text_color(cls, data, route_id):
         if route_id == 'AMTK':
             return '#000000'
         return data['FORECOLOR']
+    
+    @classmethod
+    def __get_route_icon(cls, destination):
```
Comment (joshverma) [inline gtfs_realtime_translators/translators/njt_rail.py:188]:
```suggestion
    def __get_route_icon(cls, headsign):
```

**Review by joshverma — CHANGES_REQUESTED** [review]

**Comment by joshverma** [comment]
Do you also mind cleaning up the description? `Add route icon field in intersection proto update the bindings and other modules`

File: gtfs_realtime_translators/bindings/intersection_pb2.py (hunk 1 above)
Comment (prafullint) [reply gtfs_realtime_translators/bindings/intersection_pb2.py:6]: yes

**Review by joshverma — APPROVED** [review]
Looks good, let's hold off on merging until we test it though.

**Review by JorgeBillini — APPROVED** [review]

File: gtfs_realtime_translators/bindings/gtfs-realtime.proto [hunk 8]
```diff
@@ -663,3 +663,8 @@ message TranslatedString {
   // modifications to the spec.
   extensions 1000 to 1999;
 }
+
+// Add list field type for route icon
+message RouteIconString {
```
Comment (joshverma) [inline gtfs_realtime_translators/bindings/gtfs-realtime.proto:668]: This should be removed.

**Review by joshverma — CHANGES_REQUESTED** [review]

**Review by Dara-Sy — APPROVED** [review]

**Pull request merged**